### PR TITLE
Add IO for writing to local file system

### PR DESCRIPTION
**Pull request by javajawa**
The IOConfig only supplies an Output, which can create and delete files, including writing or appending content.

### CODE REVIEW

**Comment by javajawa** [comment]
I found some notes:
- Add test for creating dirs
- Add test for binary write
- Add test for attempting to write to `../` or `/abs` style paths
- Add test for non-existent output dir
- Add test for non-initialised output dir

File: src/mewbot/io/file_storage.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+@dataclasses.dataclass
+class CreateDirectoryOutputEvent(FSOutputEvent):
+    """Creates a directory at the given path (and any parent directories)"""
+
+
+@dataclasses.dataclass
+class WriteToFileOutputEvent(FSOutputEvent):
+    """
+    Writes the given contents to the specified file.
+
```
Comment (ajCameron) [inline src/mewbot/io/file_storage.py]: Suggest adding something like "within the designated _output_path" - just to emphasize that arbitary file creation is not supported.

File: src/mewbot/io/file_storage.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    file_contents: Union[bytes, str]
+    append: bool = False
+    may_overwrite: bool = True
+
+
+@dataclasses.dataclass
+class DeleteFileOutputEvent(FSOutputEvent):
+    """
+    Deletes a file with the given name.
+    """
```
Comment (ajCameron) [inline src/mewbot/io/file_storage.py]: Again suggest adding "inside the designated _output_path"

**Review by ajCameron — DISMISSED** [review]

**Comment by ajCameron** [comment]
All looks good apart from some very minor doc suggestions.

**Comment by exachixkitsune** [comment]
Does this pull remain valid, and/or want to wait until #120 is done to comply with that system?

**Comment by javajawa** [comment]
> Does this pull remain valid, and/or want to wait until #120 is done to comply with that system?

Merging will wait for whatever change actually implements the plugin system (that's where my series of merges are slowly approaching base off of #127), but moving it to use the plugin system is pretty simple.

**Comment by ajCameron** [comment]
What the...

**Comment by ajCameron** [comment]
Okay. Name clash between a remote branch and a local branch.

Annoying. But fixable.

**Comment by javajawa** [comment]
Handled in https://github.com/mewbotorg/mewbot-io-file-storage/pull/3

**Pull request closed without merging**